### PR TITLE
Allow non authenticated users set comment author nickname

### LINKED ISSUE #208: Allow anonymous users to give nickname when commenting

### PR DESCRIPTION
**Pull request by tuomas777**
Closes #208

### CODE REVIEW

File: src/components/BaseCommentForm.js [hunk 1]
```diff
@@ -69,11 +75,22 @@ class BaseCommentForm extends React.Component {
...
     if (!this.state.collapsed) {
       return (<div className="comment-form">
         <form>
           <h3><FormattedMessage id="writeComment"/></h3>
           <Input type="textarea" value={this.state.commentText} onChange={this.handleTextChange.bind(this)}/>
+          {canSetNickname ? <h3><FormattedMessage id="nickname"/></h3> : null}
+          {canSetNickname ? (
+            <Input
+              type="text"
+              placeholder={this.props.intl.formatMessage({id: "anonymous"})}
```
Comment (akx) [inline src/components/BaseCommentForm.js]: Missing translation for "anonymous"?

**Comment by akx** [comment]
So uhm, are authenticated users shown somehow differently in the UI? That is, if I post as an authenticated "Aarni Koskela", and then some joker goes and uses "Aarni Koskela" as their nickname to post something nasty, will the audience be able to see that these are probably not the same people?

**Comment by Rikuoja** [comment]
An excellent point right there. Obviously, authenticated users will need to be differentiated in the UI. Any suggestions on what would be the user-friendly way to show the difference?

File: src/components/BaseCommentForm.js (hunk 1 above)
Comment (tuomas777) [inline src/components/BaseCommentForm.js]: https://github.com/City-of-Helsinki/kerrokantasi-ui/blob/master/src/i18n/fi.json#L41

**Comment by tuomas777** [comment]
Yeah that would be good. It needs some work on the backend too, because I don't think this information is currently present there either

**Comment by JsseL** [comment]
Maybe some kind of user icon next to the nickname for logged in users?

**Comment by Rikuoja** [comment]
Yeah, though the icon should be kinda self-evident, otherwise people will think it's the same user nevertheless. Perhaps an added "registered user" tag or something?

**Comment by tuomas777** [comment]
Also if authenticated users cannot change their nicknames, probably non authenticated users should not be able to use those exact nicknames?

**Comment by akx** [comment]
Unassigned myself pending the changes in the comments.

**Comment by tuomas777** [comment]
Updated, added registered user differentiating when showing comments.

**Comment by tuomas777** [comment]
kommenttiboksi

**Comment by tuomas777** [comment]
kommentit

**Pull request merged**